src/Pages/RegisterPage/RegisterPage.jsx

```diff
@@ -0,0 +1,194 @@
+        throw "Username may only contain letters, numbers, and @/./+/-/_ characters"
+    }
+}
+
+function validateEmail(email)
+{
+    if( !/\S+@\S+\.\S+/.test(email)){
+        throw "Invalid email address"
+    }
+}
+
+function validateInput(username, email, password){
+    if( username === ""){
+        throw "Please enter a username"
+    }
+    if( email === ""){
+        throw "Please enter an email"
+    }
+    validateEmail(email)
+    validateUsername(username)
+    validatePassword(password)
+}
+
+const register = async (data) => {
+    const url = "https://wasteless-backend.herokuapp.com/register/";
+    const headers = {
+        "Content-Type": "application/json"
+    };
+
+    const res = await fetch(url, {
+        headers,
+        method: "POST",
+        body: JSON.stringify(data)
+    });
+
+    if (res.status === 201) {
+        return await res.json();
+    } else {
+        let response = await res.text()
+        console.log(response)
+        if(response === "{\"username\":[\"A user with that username already exists.\"]}"){
+            throw "This username is already taken"
+        }
+        if(response === "{\"username\":[\"Enter a valid username. This value may contain only letters, numbers, and @/./+/-/_ characters.\"]}"){
+            throw "Username may only contain letters, numbers, and @/./+/-/_ characters"
+        }
+        if(response === "{\"email\":[\"Enter a valid email address.\"]}"){
+            throw "Invalid email address"
+        }
+        throw "An unexpected error occured"
+    }
+}
+
+const useStyles = makeStyles((theme) => ({
+    paper: {
+        marginTop: theme.spacing(8),
+        display: 'flex',
+        flexDirection: 'column',
+        alignItems: 'center',
+    },
+    avatar: {
+        margin: theme.spacing(1),
+        backgroundColor: theme.palette.primary.main,
+    },
+    form: {
+        width: '100%', // Fix IE 11 issue.
+        marginTop: theme.spacing(1),
+        padding: '7%'
+    },
+    submit: {
+        margin: theme.spacing(3, 0, 2),
+    },
+    card: {
+        boxShadow: '0 1px 1px 1px lightblue'
+    },
+}));
+
+const RegisterPage = () => {
+    const classes = useStyles();
+    const [username, setUsername] = useState("");
+    const [password, setPassword] = useState("");
+    const [email, setEmail] = useState("");
+    const [redirect, setRedirect] = useState(undefined);
+    const [disabled, setDisabled] = useState(false);
+    const [validated, setValidated] = useState("false");
+    const alertC = useRef(useContext(AlertContext));
+
+    const handleSubmit = async (e) => {
+        e.preventDefault();
+        try {
+            validateInput(username,email,password)
+            let response = await register({username: username, email: email, password: password});
+            alertC.current.showAlert("Registered successfully!", "success");
+            console.log(response)
+            setRedirect(path_list.LOGIN.route);
```

Comment:
Czy po rejestracji nie powinien ktoś się zalogować od razu? Wydaje mi się że tak ma to większy sens